src/App.tsx

```diff
@@ -1,223 +1,112 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+  useCallback,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Todo } from './types/Todo';
+import { Error } from './types/Error';
+import { getTodos } from './api/todos';
+import { TodoList } from './components/TodoList';
+import { NewTodo } from './components/NewTodo';
+import { TodosFilter } from './components/TodoFilter';
+import { ErrorNotification } from './components/ErrorNotification';
+import { TodoStatus } from './types/TodoStatus';
 
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>(todos);
+  const [hasError, setHasError] = useState<Error>({
+    message: '',
+    status: false,
+  });
+  const [todoStatus, setTodoStatus] = useState<TodoStatus>(TodoStatus.All);
+
+  const showError = useCallback((message: string) => {
+    setHasError({ status: true, message });
+
+    setTimeout(() => {
+      setHasError({ status: false, message: '' });
+    }, 3000);
+  }, []);
+
+  const loadTodos = useCallback(async () => {
+    try {
+      if (user) {
+        const todosFromServer = await getTodos(user.id);
+
+        setTodos(todosFromServer);
+      }
+    } catch (err) {
+      showError('Unable to load todos');
+    }
+  }, []);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
+
+    loadTodos();
+  }, []);
+
+  useEffect(() => {
```

Comment:
useEffect is redundant here
if you'll create a common variable `visibleTodos` it will update on every state change (if `todos` or `todoStatus` changed)

if your goal was to avoid unnecessary defining of visible todos when `hasError` state changed, use `useMemo` instead